librz/asm/asm.c

```diff
@@ -321,7 +321,7 @@ RZ_API bool rz_asm_use(RzAsm *a, const char *name) {
 		return true;
 	}
 	rz_list_foreach (a->plugins, iter, h) {
-		if (!strcmp(h->name, name) && h->arch) {
+		if (h && h->arch && h->name && !strcmp(h->name, name)) {
```

Comment:
```suggestion
		if (h->arch && h->name && !strcmp(h->name, name)) {
```
I think `h` should not be checked. If `h` is NULL there's definitely something "wronger" somewhere else.

Reply:
We can add `rz_return_if_fail()` then here.

Reply:
I would just remove that completely. If you have doubts, have a look at the rz_list_append that append plugins to `a->plugins`.

Reply:
Removed NULL check for `h` here.